debian/landscape-sysinfo.wrapper

```diff
@@ -1,17 +1,39 @@
 #!/bin/sh
-# pam_motd does not carry the environment
-[ -f /etc/default/locale ] && . /etc/default/locale
-export LANG
-cores=$(grep -c ^processor /proc/cpuinfo 2>/dev/null)
-[ "$cores" -eq "0" ] && cores=1
-threshold="${cores:-1}.0"
-if [ $(echo "`cut -f1 -d ' ' /proc/loadavg` < $threshold" | bc) -eq 1 ]; then
-    echo
-    echo -n "  System information as of "
-    /bin/date
-    echo
-    /usr/bin/landscape-sysinfo
-else
-    echo
-    echo " System information disabled due to load higher than $threshold"
+
+# do try refresh this more than once per minute
+# Due to cpu consumption and login delays (LP: #1893716)
+stamp="/var/lib/landscape/landscape-sysinfo.cache"
+NEED_UPDATE="FALSE"
+find $stamp -newermt 'now-1 minutes' 2> /dev/null | grep -q -m 1 '.' || NEED_UPDATE="TRUE"
+# If the last report in cache wasn't useful (load was too high) still wait at least 5 seconds
+if grep -q "System information disabled" $stamp 2> /dev/null; then
+    find $stamp -newermt 'now-5 seconds' 2> /dev/null | grep -q -m 1 '.' || NEED_UPDATE="TRUE"
```

Comment:
The grep for "disabled" will hopefully mostly resolve to false and thereby skip the inner check.
That allows us to to mostly use the slower 1 minute frequency (saving load).

But for the case it does have a "disabled" message in the current state cache, give it a faster (but not the "no-limits fast" of today) chance to generate some useful info.

Reply:
Outch. Please excuse my previously terrible grammar.

My point is the lines 9 to 11 appear superfluous to me.

The script never overwrites valid data with "System information disabled". It preserves the last "good" data due to the check on line 30. This is fine.
It means that, unless the first sysinfo-wrapper run occurs during high load, there will be good data. Otherwise load will just preserve the last data.

That also means the grep (line 9) will practically never match, whatever the load is. Making this check somewhat useless.
In the very very unlikely even that the data is bad (i.e. info disabled), the find (line 10) will practically never match as you never alter the modified the timestamp under load. So `NEED_UPDATE` will practically always end up being set TRUE if there is bad data and high load. Thus, the only time where line 9-11 will ever wait 5 seconds is if installing the package under high load. After that, the "now-5 seconds" condition will never ever match.

That's what I mean by those lines being superfluous.

Reply:
Thanks for the extended explanation of your thought @simpoir !

Originally I added this to prevent to be a problem for a  system with continuous high load - especially as we'd add more to it. I was able to better follow your thoughts now and agree, except with the suggested solution.

In a perma-busy system with no good data yet what will happen:
1. the check in line 9 finds the old data as "disabled"
2. the check in line 10 will see it is old and evaluate to `NEED_UPDATE="TRUE"`
3. a load check in line 17-20 will spot that we are still too busy
4. as the old info isn't a good one yet we will update the stamp with a new "disabled"
5. on the next call the check in line 10 will hold back further updates for 5 seconds.

Without that line 10 check, in those unlikely but possible permanent-busy scenarios, we'd potentially do way too many load checks. Therefore I do not like the suggested solution, but OTOH you are also right we are replacing one (load) check with the another (age) which isn't great either.

Thinking about it more I'm tempted to say "whatever it is let us what a minute" which would mean removing lines 8-11 entirely. That would be less code and less consumption.
@simpoir what would you think about that?

Reply:
Hi @simpoir  - I experimented with this (and other things as I have more PRs up and also got other feedback).
I've found:

- a faster (and slightly more readable) way to do the checks for the old files
- a slightly faster and much more readable way to print the inner statements
- dropping the 5s check helps (as expected, mostly due to skipping the initial check for disabled)
 
I'll need to test those again which takes a while, but I've added them as individual commits (for better readability) here already. Eventually we can squash them (or not - up to you).

Reply:
FYI - I won't get to it before tomorrow - but testing found that I need to update it again slightly for the case os "no cache file present"

Reply:
Ok I now fixed and tested this all over again.
- The speed/consumption gains are all still good
- The ouput layout is equal to before
- I have resolved the former discussions brought up here.
- ensured we get no "non empty dir" warning on purge

Ready for re-review to hopefully accept and upload it early next week (I'd do so once we agree here).
I'll close this conversation as I think all is resolved and will hit the re-review button ...